backend/src/service/storeService.js

```diff
@@ -0,0 +1,48 @@
+const storeModel = require("../model/storeModel")
+const geoService = require("../service/geoService")
+const moment = require('moment-timezone');
+//assume that a store takes an average of 20 minutes (1200 sec) to prepare a purchase
+const prapareTime = 1200;
+
+
+module.exports = {
+    async initdb() {
+        const storeLength = await storeModel.initDb()
+        return storeLength;
+    },
+    async getClosest(location) {
+        const { latitude, longitude } = location;
+        const limit = geoService.getGeoHashLimits(latitude, longitude, 5);
+
+        let store = await storeModel.getClosest(limit);
+        if (store) {
+            const distance = await this.getDistance(
+                { lat: latitude, lng: longitude },
+                { lat: store.latitude, lng: store.longitude }
+            )
+            //5 mins per km
+            const trafficTime = 300 * distance.distance / 1000
+            const delay =  Math.random() * (2000 - 1200) + 1200
+
+            bestDeliveryTime = moment(new Date()).tz('America/Bogota').add(distance.duration + prapareTime + trafficTime, 'seconds')
```

Comment:
Can you imagine a better way to manage the timezone? for example the stores that I sent are in Monterrey, and Monterrey change from DST to STD, which means that they are going to be 1 hour ahead of us in October?
Also, why do you need to use new Date() with moment?

Reply:
There is no need to use new Date() in this case (I changed it), in the last commit  I used a google api using the latitude and longitude I already got with the address to find the time zone